sched/task/task_exithook.c

```diff
@@ -51,53 +51,7 @@
  *
  ****************************************************************************/
 
-#if defined(CONFIG_SCHED_ATEXIT) && !defined(CONFIG_SCHED_ONEXIT)
-static inline void nxtask_atexit(FAR struct tcb_s *tcb)
-{
-  FAR struct task_group_s *group = tcb->group;
-
-  /* Make sure that we have not already left the group.  Only the final
-   * exiting thread in the task group should trigger the atexit()
-   * callbacks.
-   *
-   * REVISIT: This is a security problem In the PROTECTED and KERNEL builds:
-   * We must not call the registered function in supervisor mode!  See also
-   * on_exit() and pthread_cleanup_pop() callbacks.
-   *
-   * REVISIT:  In the case of task_delete(), the callback would execute in
-   * the context the caller of task_delete() cancel, not in the context of
-   * the exiting task (or process).
-   */
-
-  if (group && group->tg_nmembers == 1)
-    {
-      int index;
-
-      /* Call each atexit function in reverse order of registration atexit()
-       * functions are registered from lower to higher array indices; they
-       * must be called in the reverse order of registration when the task
-       * group exits, i.e., from higher to lower indices.
-       */
-
-      for (index = CONFIG_SCHED_EXIT_MAX - 1; index >= 0; index--)
-        {
-          if (group->tg_exit[index].func.at)
-            {
-              atexitfunc_t func;
-
-              /* Nullify the atexit function to prevent its reuse. */
-
-              func = group->tg_exit[index].func.at;
-              group->tg_exit[index].func.at = NULL;
-
-              /* Call the atexit function */
-
-              (*func)();
-            }
-        }
-    }
-}
-#else
+#ifdef CONFIG_SCHED_ATEXIT
 #  define nxtask_atexit(tcb)
```

Comment:
I don't understand this.  If CONFIG_SCHED_ATEXIT is defined, then nxtask_atexit() is a defined out?  I think you need it.